added capacity expansion design to documentation (#304)
open-energy-transition/pypsa-zambia: doc/modelling_note.md

Once a validation run completed, the outputs are available in `pypsa-earth-status/results`.

For more advanced analysis (e.g. checking imports and exports), a validation notebook is available in [notebooks](https://github.com/open-energy-transition/pypsa-zambia/tree/main/notebooks) folder.

## Capacity Expansion Validation Setup

Capacity-expansion outputs are checked against Zambia's 2023 Integrated Resource Plan (IRP), using the [IRP scenarios scoping document](https://docs.google.com/document/d/13av6J_Yz-iMXanWeEhmLaMoAJUq39FRrF4-dA1Gn8Vs/edit?usp=sharing).

> Demand and costs in the model are already set to match IRP figures for each planning year. The validation is in what the model chooses to build and generate given those inputs, compared against the IRP's own plan for the same years: installed capacity by technology, generation mix, and emissions.

Reviewer: You left out the investment required for generation and transmission, as well as the transmission capacity comparison.

Comparison is done at four horizons - 2025, 2030, 2040, and 2050. Comparison plots are produced via `scripts/plot_scenario_comparison.py` and saved to `results/comparison_plots/cap_exp_zambia/`.

## Capacity Expansion

Capacity expansion scenarios extend the dispatch-only run by letting the model size new generation to meet demand at future planning horizons, using cost and technology parameters calibrated to Zambia's 2023 Integrated Resource Plan (IRP).

### Design

> **What can be expanded.** Only solar and onshore wind can be newly built by the model. Hydro, coal, oil, biomass, and geothermal capacities are fixed in each planning-year snapshot: whatever is scheduled to be in service that year (based on commissioning/decommissioning dates in the powerplants data) is what the model has to work with, and the optimizer only decides how much solar and wind to add on top.

Reviewer: In addition to adding solar and onshore wind generation capacity, the model can also expand the transmission network.

> Solar and wind siting is currently based on general land-cover suitability across the whole country, with no per-region cap and no restriction to specific sites. The IRP instead limits wind to a set of measured candidate sites and caps additions at 1,000 MW per region per planning period.

Reviewer: Do we have a reference for this?

Author: <img width="770" height="285" alt="Image" src="https://github.com/user-attachments/assets/ae9e36aa-c1a4-41f2-927e-3c1bbadd853d" />
You can fin it here in the scoping document attached [here](https://docs.google.com/document/d/13av6J_Yz-iMXanWeEhmLaMoAJUq39FRrF4-dA1Gn8Vs/edit?usp=sharing)

Reviewer 2: Could be good to make sure a document is publicly available

**Four independent planning-year snapshots.** `configs/scenarios_zambia/config.cap_exp_zambia_{2025,2030,2040,2050}.yaml` each merge on top of the shared `cap_exp_zambia_base.yaml` and are solved as fully independent optimizations - there is no myopic/multi-horizon linkage between them. A generator built in the 2040 run has no bearing on 2050; each year re-derives its own fixed fleet, demand level, and solar/wind buildout from scratch. This means that capacity trajectories between horizons are not guaranteed to be monotonic.

**Weather and hydro conditions.** All four planning years are solved against the same single weather year - ERA5 `cutout-{year}-era5`, hourly, within a given year e.g `2023-01-01` to `2024-01-01`.

> Hydro specifically is non-extendable in every scenario, normalized via the IRENA method against 2023 generation statistics with a uniform multiplier of 2.19, and drawn from the same `{year}` ERA5 cutout as solar/wind.

Reviewer: We are now using GloFAS instead of ERA5, so the 2.19 multiplier is no longer valid.

Author: @tacwebservices in the `cap_exp_zambia_base.yaml` the `renewable:hydro` uses an ERA5 cutout with a `multiplier: 2.19`

**Demand and cost basis.** Demand is a DemandCast hourly profile for the 2023 weather year, scaled per planning year (`load_options:scale`) to match IRP Table 3's national demand projection for that year (e.g. 2030: scale 2.64 against IRP's 41,925 GWh vs. DemandCast's unscaled 15,909 GWh base). Investment, marginal cost, FOM, lifetime, and CO2 emission factors are transcribed directly from the Zambia 2023 IRP (Table 5, Table 9, Annex 3), converted at a fixed 0.7532 EUR/USD rate, with `costs.year` set per scenario.

### Results

Generation mix by technology, model vs. IRP Table 12, in TWh:

| Year | Hydro Storage | Hydro RoR | Solar | Wind | Coal & Oil | Biomass & Geo | Total |
|---|---|---|---|---|---|---|---|
| 2025/26 | 11.4 / 9.2 | 10.1 / 9.4 | 0.3 / 3.0 | 0.1 / 0.9 | 0.9 / — | — / — | 22.8 / 22.9 |
| 2030 | 18.9 / 9.7 | 11.2 / 20.8 | 4.0 / 3.8 | 2.8 / 2.5 | 2.9 / 4.0 | 1.8 / 1.0 | 41.6 / 41.9 |
| 2040 | 18.2 / 10.0 | 21.6 / 23.0 | 5.5 / 6.4 | 3.8 / 7.0 | 3.0 / 7.0 | 7.3 / 6.1 | 59.3 / 59.6 |
| 2050 | 19.0 / 8.3 | 25.7 / 31.4 | 6.9 / 9.2 | 3.6 / 11.5 | 4.4 / 5.4 | 15.8 / 9.9 | 75.5 / 75.7 |

*(model / IRP)*

> Total demand matches IRP closely at every horizon, as expected since it's calibrated to it directly. Wind is under-represented relative to the IRP throughout, and the gap grows with each horizon, while hydro storage carries a larger share of generation than the IRP assumes. These figures are from an earlier scenario run, using an earlier version of the wind/solar siting data; expect the mix to shift as the model is re-run with updated inputs.

Reviewer: It would be good to mention that we have discrepancies in the wind results due to the low capacity factors.

Once a validation run completed, the outputs are available in `pypsa-earth-status/results`.

For more advanced analysis (e.g. checking imports and exports), a validation notebook is available in [notebooks](https://github.com/open-energy-transition/pypsa-zambia/tree/main/notebooks) folder.

## Capacity Expansion Validation Setup

> Capacity-expansion outputs are checked against Zambia's 2023 Integrated Resource Plan (IRP), using the [IRP scenarios scoping document](https://docs.google.com/document/d/13av6J_Yz-iMXanWeEhmLaMoAJUq39FRrF4-dA1Gn8Vs/edit?usp=sharing).

Reviewer 2: I think we can be a bit more explicit to increase clarity:

```suggestion
Capacity-expansion outputs are checked against Zambia's 2023 Integrated Resource Plan (IRP). A capacity expansion model has been setup using IRP assumptions summarised the [IRP scenarios scoping document](https://docs.google.com/document/d/13av6J_Yz-iMXanWeEhmLaMoAJUq39FRrF4-dA1Gn8Vs/edit?usp=sharing).
```

> Demand and costs in the model are already set to match IRP figures for each planning year. The validation is in what the model chooses to build and generate given those inputs, compared against the IRP's own plan for the same years: installed capacity by technology, generation mix, investment in generation and transmission, transmission capacity, and emissions.

Reviewer 2: IRP doesn't account for learning effects for costs. I'd also avoid duplication in explaining that validation implies comparison of modeling outputs against references and focus on a list of the parameters:

```suggestion
Demand in the model is adjusted to match IRP figures for each planning year. The validation includes against the IRP's own plan for the same years for installed capacity by technology, generation mix, investment in generation and transmission, transmission capacity, and emissions.
```

> Comparison is done at four horizons - 2025, 2030, 2040, and 2050. Comparison plots are produced via `scripts/plot_scenario_comparison.py` and saved to `results/comparison_plots/cap_exp_zambia/`.

Reviewer 2: I'd highlight that we compare against scenarios, not against historical horizons or forecasts. Also, I'd suggest to avoid using hyphens unless there are very good reasons for that.

```suggestion
Comparison is done for IRP scenarios corresponding to the following four horizons: 2025, 2030, 2040, and 2050. Comparison plots are produced via `scripts/plot_scenario_comparison.py` and saved to `results/comparison_plots/cap_exp_zambia/`.
```

**Note:** transmission isn't covered by the comparison tooling yet. `scripts/plot_scenario_comparison.py` currently excludes lines from both the installed-capacity and investment figures, so those only reflect generation and storage. The model itself does expand transmission.

## Capacity Expansion

> Capacity expansion scenarios extend the dispatch-only run by letting the model size new generation to meet demand at future planning horizons, using cost and technology parameters calibrated to Zambia's 2023 Integrated Resource Plan (IRP).

Reviewer 2: This part must be moved just before `Capacity Expansion Validation Setup` section to ensure a smooth flow

### Design

> **What can be expanded.** Only solar and onshore wind can be newly built by the model. Hydro, coal, oil, biomass, and geothermal capacities are fixed in each planning-year snapshot: whatever is scheduled to be in service that year (based on commissioning/decommissioning dates in the powerplants data) is what the model has to work with, and the optimizer only decides how much solar and wind to add on top.

Reviewer 2: Also dispatch is optimized:

```suggestion
**What can be expanded.** Only solar and onshore wind can be newly built by the model. Hydro, coal, oil, biomass, and geothermal capacities are fixed in each planning-year snapshot: whatever is scheduled to be in service that year (based on commissioning/decommissioning dates in the powerplants data) is what the model has to work with, and the optimizer only decides how to dispatch the available capacity, and how much solar and wind to add on top.
```

Solar and wind siting is currently based on general land-cover suitability across the whole country, with no per-region cap and no restriction to specific sites. The IRP instead limits wind to a set of measured candidate sites and caps additions at 1,000 MW per region per planning period.

> Without those limits, the model can build wind capacity far beyond what's realistic, at very low realized capacity factors.

Reviewer 2: There are two points mixed here:
1) our model has lower capacity factors for wind and solar as compared with IRP
2) IRP sets expansion caps to represent system stability limitations

I think the easiest way to deal with that is to remove this fragment but feel free to suggest revisions if you prefer.
```suggestion
```

> **Four independent planning-year snapshots.** `configs/scenarios_zambia/config.cap_exp_zambia_{2025,2030,2040,2050}.yaml` each merge on top of the shared `cap_exp_zambia_base.yaml` and are solved as fully independent optimizations - there is no myopic/multi-horizon linkage between them. A generator built in the 2040 run has no bearing on 2050; each year re-derives its own fixed fleet, demand level, and solar/wind buildout from scratch. This means that capacity trajectories between horizons are not guaranteed to be monotonic.

Reviewer 2: Let's keep it simpler and more consistent with the existing termininology:
```suggestion
**Four independent planning-year snapshots.** `configs/scenarios_zambia/config.cap_exp_zambia_{2025,2030,2040,2050}.yaml` each merge on top of the shared `cap_exp_zambia_base.yaml` and are solved as an overnight optimisation. A generator built in the 2040 run has no bearing on 2050 and each year obtains its own fixed fleet, demand level, and solar/wind buildout from scratch. This means that capacity trajectories between horizons are not meant to be monotonic.
```

> **Weather and hydro conditions.** All four planning years are solved against the same single weather year - ERA5 `cutout-{year}-era5`, hourly, within a given year e.g `2023-01-01` to `2024-01-01`.

Reviewer 2: We need to review the raw output, otherwise the text doesn't make sense.

```suggestion
**Weather and hydro conditions.** All four planning years are solved against the same single weather year using ERA5-originated weather archive `cutout-{year}-era5` in hourly resolution.
```

> Hydro specifically is non-extendable in every scenario, normalized via the IRENA method against 2023 generation statistics with a uniform multiplier of 2.19, and drawn from the same `{year}` ERA5 cutout as solar/wind.

Reviewer 2: @Mwiche96 you are right for the mainstream version of the model. However, we have been used GloFAS data for capacity expansion runs as ERA5-based modelling outputs are of very poor quality. Given high complexity of hydro topic, I'd leave it for a dedicated discussion in a dedicated discussion board #154 and mention only the major points.

```suggestion
Hydro can be represented using either ERA5-derived runoff calculated by `atlite` procedure, or a novel method based of GloFAS data. Expansion of hydro is currently accounted for by considering development plans for installed hydro generation capacity.
```

> Solar and wind use `correction_factor` 0.93 and 0.83 respectively (empirical de-rating of the raw ERA5-derived profile), and siting eligibility is computed with a `simple` land-use potential method.

Reviewer 2: I think explanations must be given for both correction factors and `simple` method, otherwise this fragment sounds confusing.  Let's open an issue to document wind and solar properly, and remove it for now from the doc?

```suggestion
```

> **Demand and cost basis.** Demand is a DemandCast hourly profile for the 2023 weather year, scaled per planning year (`load_options:scale`) to match IRP Table 3's national demand projection for that year (e.g. 2030: scale 2.64 against IRP's 41,925 GWh vs. DemandCast's unscaled 15,909 GWh base). Investment, marginal cost, FOM, lifetime, and CO2 emission factors are transcribed directly from the Zambia 2023 IRP (Table 5, Table 9, Annex 3), converted at a fixed 0.7532 EUR/USD rate, with `costs.year` set per scenario.

Reviewer 2: Let's remove details which are not relevant for the design description and can be taken from the codebase if needed:

```suggestion
**Demand and cost basis.** Demand is a DemandCast hourly profile for the selected 2023 weather year, scaled per planning year (`load_options:scale`) to match IRP Table 3's national demand projection for that year. Investment, marginal cost, FOM, lifetime, and CO2 emission factors are transcribed directly from the Zambia 2023 IRP (Table 5, Table 9, Annex 3), converted at a fixed 0.7532 EUR/USD rate.
```

### Results

Generation mix by technology, model vs. IRP Table 12, in TWh:

| Year | Hydro Storage | Hydro RoR | Solar | Wind | Coal & Oil | Biomass & Geo | Total |
|---|---|---|---|---|---|---|---|
| 2025/26 | 11.4 / 9.2 | 10.1 / 9.4 | 0.3 / 3.0 | 0.1 / 0.9 | 0.9 / — | — / — | 22.8 / 22.9 |
| 2030 | 18.9 / 9.7 | 11.2 / 20.8 | 4.0 / 3.8 | 2.8 / 2.5 | 2.9 / 4.0 | 1.8 / 1.0 | 41.6 / 41.9 |
| 2040 | 18.2 / 10.0 | 21.6 / 23.0 | 5.5 / 6.4 | 3.8 / 7.0 | 3.0 / 7.0 | 7.3 / 6.1 | 59.3 / 59.6 |
| 2050 | 19.0 / 8.3 | 25.7 / 31.4 | 6.9 / 9.2 | 3.6 / 11.5 | 4.4 / 5.4 | 15.8 / 9.9 | 75.5 / 75.7 |

*(model / IRP)*

> Total demand matches IRP closely at every horizon, as expected since it's calibrated to it directly. Wind is under-represented relative to the IRP throughout, and the gap grows with each horizon, while hydro storage carries a larger share of generation than the IRP assumes. These figures are from an earlier scenario run, using an earlier version of the wind/solar siting data; expect the mix to shift as the model is re-run with updated inputs.

Reviewer 2: Could be good to validation outputs which may be best to be presented as a live discussion board since we still have some ideas on how to improve accuracy of the output
